onnxruntime/core/framework/session_state.cc

```diff
@@ -1030,7 +1030,10 @@ Status SessionState::CreateSubgraphSessionState() {
   for (auto& node : graph_.Nodes()) {
     for (auto& entry : node.GetAttributeNameToMutableSubgraphMap()) {
       const auto& ep = node.GetExecutionProviderType();
-      if (!ep.empty() && ep != kCpuExecutionProvider && ep != kCudaExecutionProvider && ep != kRocmExecutionProvider && ep != kDmlExecutionProvider) {
+      if (!ep.empty() &&
```

Comment:
Basicaly tells ORT core framework that JSEP allows subgraph operators - it forces the core framework to collect metadata (like tensor device placement needed for subsequent copy operations in subgraph based ops)